universal-application-tool-0.0.1/test/services/applicant/ApplicantDataTest.java

```diff
@@ -79,6 +79,16 @@ public void readString_findsCorrectValue() throws Exception {
 
     assertThat(found).hasValue("orange");
   }
+  
+  @Test
+  public void readInteger_findsCorrectValue() throws Exception {
+    String testData = "{ \"applicant\": { \"age\": 30 } }";
+    ApplicantData data = new ApplicantData(testData);
+
+    Optional<Integer> found = data.readInteger(Path.create("applicant.age"));
+
+    assertThat(found).hasValue(30);
+  }
```

Comment:
Suggestion (blocking): please add a test that passes a path to a bad value that isn't present and asserts that it returns an empty optional.